Lib/test/test_gdb.py

```diff
@@ -17,12 +17,18 @@
 
 def get_gdb_version():
     try:
-        proc = subprocess.Popen(["gdb", "-nx", "--version"],
+        cmd = ["gdb", "-nx", "--version"]
+        proc = subprocess.Popen(cmd,
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE,
                                 universal_newlines=True)
         with proc:
-            version = proc.communicate()[0]
+            version, stderr = proc.communicate()
+
+        if proc.returncode:
+            raise Exception(f"Command {' '.join(cmd)!r} failed "
+                            f"with exit code {proc.returncode}: "
+                            f"stdout={version!r} stderr={stderr!r}")
```

Comment:
The test is normally skipped if gdb is not found:

```python
raise unittest.SkipTest("Couldn't find gdb on the path")
```

Should we not do the same?

Reply:
The test is still skipped when gdb is not found. But the new code is when gdb is available but is broken. I would prefer to not fail silently when gdb is broken, but get an error even if it's a third-party component. It should help to debug Python CIs like buildbots.

It's like the existing test which raises an exception if Python fails to parse the gdb version:

```
    match = re.search(r"^GNU gdb.*?\b(\d+)\.(\d+)", version)
    if match is None:
        raise Exception("unable to parse GDB version: %r" % version)
```